app/assets/javascripts/form-kit/addon/components/control/input.gjs

```diff
@@ -0,0 +1,67 @@
+import Component from "@glimmer/component";
+import { assert } from "@ember/debug";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+
+const SUPPORTED_TYPES = [
+  "color",
+  "date",
+  "datetime-local",
+  "email",
+  "hidden",
+  "month",
+  "number",
+  "password",
+  "range",
+  "search",
+  "tel",
+  "text",
+  "time",
+  "url",
+  "week",
+];
+
+export default class FKControlInput extends Component {
+  constructor(owner, args) {
+    super(...arguments);
+
+    assert(
+      `input component does not support @type="${args.type}" as there is a dedicated component for this. Please use the \`field.${args.type}\` instead!`,
+      args.type === undefined || !["checkbox", "radio"].includes(args.type)
+    );
+
+    assert(
+      `input component does not support @type="${
+        args.type
+      }", must be one of ${SUPPORTED_TYPES.join(", ")}!`,
+      args.type === undefined || SUPPORTED_TYPES.includes(args.type)
+    );
```

Comment:
BTW, these Ember assertions are optimised out of production builds. So if people are developing themes against production sites, they will never be hit. Maybe fine, but just want to mention.

Reply:
Yes I know how it works but indeed I didnt think of people developing themes directly against a production build. That was actually a question on my list to ask you why we don't use assert more, I guess I know now :p